### PR TITLE
Power9 fix p2

### PR DESCRIPTION
**Pull request by munroesj52**
The second part. With more P9 enabled and doxygen updates (but not enabled yet). Also some tests are deferred until the restructuring in part 3.

### CODE REVIEW

File: src/vec_bcd_ppc.h [hunk 1]
```diff
@@ -147,55 +155,39 @@ vec_bcdadd (vui32_t a, vui32_t b)
             : "v" (a),
               "v" (b)
             : "cr6" );
-#else
-#if 0
```
Comment (ThinkOpenly) [inline src/vec_bcd_ppc.h]: curious what's going on here. is this also a temporary disable pending infrastructure?

File: src/vec_bcd_ppc.h [hunk 2]
```diff
@@ -147,55 +155,39 @@ vec_bcdadd (vui32_t a, vui32_t b)
...
 #else
         _Decimal128 d_t;
         d_t = vec_BCD2DFP (a) + vec_BCD2DFP (b);
-#endif
         t = vec_DFP2BCD(d_t);
 #endif
         return (t);
 }
 
-/** \brief Subtract two Vector Signed BCD 31 digit values.
```
Comment (ThinkOpenly) [inline src/vec_bcd_ppc.h]: did you change the order of these functions? makes it tough to review.  I presume there were no code changes.

**Comment by munroesj52** [comment]
On Wed, 2018-06-06 at 07:38 -0700, Paul Clarke wrote:
> @ThinkOpenly commented on this pull request.
> In src/vec_bcd_ppc.h:
> > @@ -147,55 +155,39 @@ vec_bcdadd (vui32_t a, vui32_t b)
>              : "v" (a),
>                "v" (b)
>              : "cr6" );
> -#else
> -#if 0
> curious what's going on here. is this also a temporary disable
> pending infrastructure?
The bcdadd/bcdsub will alwayts set the condition code 6 (even if it is
not used in the code). The compiler needs to know this in case it wants
to use this a CR during instruction scheduling.


> In src/vec_bcd_ppc.h:
> >          t = vec_DFP2BCD(d_t);
>  #endif
>          return (t);
>  }
>  
> -/** \brief Subtract two Vector Signed BCD 31 digit values.
> did you change the order of these functions? makes it tough to
> review. I presume there were no code changes.
Yes, as we discussed. I knew this would complicate the review but you
seemed to ok with that.

Anyway at this point it would be really hard to reverse the reordering.
Might be easier to just review the new code for any functions
with _ARCH_PWR9.

Otherwise it is mostly removing debug code (#if 0, and printf) and
replacing long strings of inline _asm with equivalent altivec.h built-
in sequences.

**Comment by munroesj52** [comment]
Cnd we move on this? 

Most of these changes where reviewed for power8-mul fork which was not pushed because it was too big. 

These changes only look complicated when compared to pveclib master but these changes have already been reviewed.

You can see the actual differences by dirdiff between in the power8-mul and power9-fix-p2 forks.

specifically:

        * src/vec_bcd_ppc.h: Add doxygent brief description.
        (vec_bcdctb100s), New Function.
        * src/vec_char_ppc.h: Add doxygent brief description.
        * src/vec_int128_ppc.h: Add doxygent brief description.
        Add PWR7 and PWR9 implementations where needed.
        Replace blocks of in-line asm with <altivec.h> intrinsics.
        (vec_clzq, vec_popcntq, vec_slqi, vec_sraqi, vec_srqi):
        New Function.

File: src/vec_bcd_ppc.h (hunk 1 above)
Comment (ThinkOpenly) [reply src/vec_bcd_ppc.h]: My comment here was not asking about cr6, but why are you hiding the code below `#if 0`?

**Comment by munroesj52** [comment]
On Mon, 2018-06-11 at 08:03 -0700, Paul Clarke wrote:
> @ThinkOpenly commented on this pull request.
> In src/vec_bcd_ppc.h:
> > @@ -147,55 +155,39 @@ vec_bcdadd (vui32_t a, vui32_t b)
>              : "v" (a),
>                "v" (b)
>              : "cr6" );
> -#else
> -#if 0
> My comment here was not asking about cr6, but why are you hiding the
> code below #if 0?

My process it to try things then I look at the generated code.

-#if 0
-        _Decimal128 d_t, d_a, d_b;
-        d_a = vec_BCD2DFP (a);
-        d_b = vec_BCD2DFP (b);
-        d_t = d_a + d_b;
-#else
         _Decimal128 d_t;
         d_t = vec_BCD2DFP (a) + vec_BCD2DFP (b);
-#endif

Since the different compilers may behave different behavior / code-gen, 
I use this trick to keep both versions around in case I need to revert.
For example if the new code does not compile correctly for older
compiler I am trying to support!

I try to clean these up before committing upstream. Looks like I missed
this one on the original commit, and cleaning up it up now.

By my training this is incidental background and not worthy of a change
log entry.

**Comment by munroesj52** [comment]
thanks

**Pull request merged**